domainobjs/ts/commands.ts

```diff
@@ -0,0 +1,366 @@
+import {
+    decrypt,
+    encrypt,
+    genRandomSalt,
+    hash4,
+    sign,
+    verifySignature,
+    Signature,
+    Ciphertext,
+    EcdhSharedKey,
+} from "maci-crypto";
+import * as assert from "assert";
+import { PubKey } from "./publicKey";
+import { PrivKey } from "./privateKey";
+import { Message } from "./message";
+
+/**
+ * @notice Base class for Commands
+ */
+export abstract class Command {
+    public cmdType: bigint;
+
+    constructor(cmdType: bigint) {
+        this.cmdType = cmdType;
+    }
+
+    public copy(): Command {
+        return this;
+    }
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    public equals(command: Command) {}
+
+    /**
+     * Serialize into a JSON object
+     */
+    public toJSON() {
+        return {
+            cmdType: this.cmdType.toString(),
+        };
+    }
+
+    /**
+     * Deserialize into a Command instance
+     * @param json
+     */
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    static fromJSON(json: any) {}
```

Comment:
Maybe it's better to get rid of static method in favour of instance one?

```suggestion
    abstract fromJSON(json: unknown): Command;
```

Reply:
I'm not sure that it makes sense for `fromJSON` to be an instance method, as usually its objective is to generate a new instance of a class. 
Perhaps we get rid of the abstract class altogether? or do you see any benefits of keeping it here?

Reply:
Ok, so in this case I'd add `throw new Error("not implemented")` for this method to prevent any calls from abstract class.
And you can use interface instead of abstract class but interfaces don't have static members. I think some parent interface or class should be to not expose implementation details (like T or P command), so we can work only with `Command`.

Reply:
Ok, good. After this PR is merged I'll start adding linter config for this package.